### DISCUSSION TITLE
Adding a picture of my garden

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#785] [Q&A] [answered]
Hello,

I have the idea of adding a picture (taken automatically every hour) of my garden or a tree on the main page of my Birdnet-Pi.

To do so I resuscitated an old Pi 3 with a webcam, installed Motioneye https://github.com/motioneye-project/motioneye/tree/dev on it so it can take a picture every hour then upload it via FTP to my Birdnet-Pi.

I thought to replace the spectrogram image below the "Currently Analyzing" label and changing the label to "My location".

Could you tell me where the image is located? Or maybe you have a better idea of how to achieve this?

Thank you.

**Comment by ehpersonal38** [comment, marked as answer]
Hi @blogdemoi. I've just whipped up something that should hopefully meet your needs. Go ahead and Update as usual, then fill out the "Custom Image" fields at the bottom of Tools -> Settings -> Advanced Settings. Hopefully you'll then be able to see your image in all its glory on the "Overview" page at the bottom.

**Reply by blogdemoi** [reply]
Thank you @ehpersonal38 ! That's a cool feature. I just added a one year old picture of a tree in my little garden :   https://birdnetpi.arnauld.org

The result is quite good. In the next few days I will finish to set up Motioneye and choose an emplacement for the webcam to take a picture at a regular interval. Or maybe just use a script (without installing Motioneye) to take pictures using the same Lepotato that is running the Birdnet-Pi... 
I will try both solutions to see what is the more convenient.

**Reply by blogdemoi** [reply]
Maybe embedding weather info in the picture can be a good idea too : https://github.com/chubin/wttr.in

**Reply by Svardsten53** [reply]
Nice feature! Please add a blank line at the bottom of the page so it doesn't look like something is missing from the last image (spectrogram or own picture). Thanks!

**Comment by blogdemoi** [comment]
So I finally tried to plug in a USB webcam in the same Lepotato running the Birdnet-Pi. I installed fswebcam (it is not possible to use the traditional camera pi commands to interact with the USB webcam as they can only be used with the official camera pi), https://www.sanslogic.co.uk/fswebcam/. 

Then I used a script to take a picture (I still have to fine tune the settings in fswebcam, color, contrast, luminosity, etc to have a nice picture).

I set up a crontab to run the script every hour, and voila, the picture is at the bottom of my Birdnet-Pi page, 
https://birdnetpi.arnauld.org/
thanks to @ehpersonal38 for adding the feature to add an image.

**Reply by blogdemoi** [reply]
The script (running every hour) I am using to take the picture, taking a picture with fswebcam and resizing, quality improvement and  making it lighter with imagemagick (most of the settings found on the web):

```
fswebcam -d /dev/video1 -D 2 -S 20 --set saturation=20% gain=50% brightness=30% --set contrast=20%  -F 1 -r  1280x960 --no-banner ~/BirdNET-Pi/homepage/images/jardi.jpg &&
convert -bordercolor Transparent -border 1x1 -filter Triangle -define filter:support=2 -thumbnail 944 -unsharp 0.25x0.25+8+0.065 -dither None -posterize 136 -quality 82 -define jpeg:fancy-upsampling=off  -interlace none -colorspace sRGB -strip ~/BirdNET-Pi/homepage/images/jardi.jpg ~/BirdNET-Pi/homepage/images/jardin.jpg
```

**Comment by blogdemoi** [comment]
When looking at the "View log" section, I have a lot of: 

> 18:20:32---/etc/birdnet/birdnet.conf: line 216: jardin: command not found

Which seems to refer in the birdnet.conf to:

> These are just for debugging
> LAST_RUN=
> THIS_RUN=
> IDFILE=/home/arnauld/BirdNET-Pi/IdentifiedSoFar.txt
> CUSTOM_IMAGE=/home/arnauld/BirdNET-Pi/homepage/images/jardin.jpg
> CUSTOM_IMAGE_TITLE=Le jardin

The last line is line 216.

**Reply by ehpersonal38** [reply]
Thanks for identifying this bug. I guess I forgot to wrap the CUSTOM_IMAGE_TITLE in quotes when you go to change it in Advanced Settings. I've just fixed that now and all you'll have to do is manually wrap quotes around `CUSTOM_IMAGE_TITLE` in your birdnet.conf, and then update your system.

**Comment by lloydbayley** [comment]
@ehpersonal38 I'm getting a placeholder even when I have no image defined. Looking at the code, each check is returning true even with nothing set, so it's displaying the empty image placeholder. I tried a !empty in there for the $config part but I'm not sure where the $_GET comes in. Was there supposed to be a tickbox to enable it? Anyway, I can't see what the problem is and instead of me struggling at 1:30am, I'll throw it over to 'he who wrote the code and knows it better than I' :) PS: It also seems to be refreshing the placeholder really, really quickly. Tested with multiple browsers.

**Reply by ehpersonal38** [reply]
Ah, good spot! The problem is that I was only checking if the config variable CUSTOM_IMAGE is empty, but since the variable is set like `CUSTOM_IMAGE=""`, the double quotes _technically_ means it's not empty. So I just did a simple if(strlen($CUSTOM_IMAGE)) > 2) { // display the image }; 

And yes the tickbox to enable this feature is in Tools -> Settings -> Advanced Settings -> Custom Image.

Thanks again for finding this and troubleshooting it!

**Reply by lloydbayley** [reply]
Sounds fair-enough! Yep, easy mistake to make :) 
Oddly, I have no tickbox, just the config fields.
<img width="1215" alt="image" src="https://user-images.githubusercontent.com/21309308/227260731-e54fbaa1-053c-49ff-87a3-50b8184c9d2d.png">

Anyway, I should NOT be still up and looking at code so I'm off to bed! 
Have a good one! :)
